### PR TITLE
add infer_batch_size to AllenNLPProcessor

### PR DESCRIPTION
**Pull request by DianLiI**
This PR fixes [#4]. 

### Description of changes
Add config option `infer_batch_size` in case there are too many sentences.

### Possible influences of this PR.
N/A

### Test Conducted
Pipeline containing AllenNLPProcessor.

### CODE REVIEW

File: forte_wrapper/allennlp/allennlp_processors.py [hunk 1]
```diff
@@ -126,33 +129,41 @@ def default_configs(cls):
...
-                    result.update(results[key][i])
-            if "tokenize" in self.configs.processors:
-                # creating new tokens and dependencies
-                tokens = self._create_tokens(input_pack, sentences[i], result)
-                if "depparse" in self.configs.processors:
-                    self._create_dependencies(input_pack, tokens, result)
-                if 'srl' in self.configs.processors:
-                    self._create_srl(input_pack, tokens, result)
+
+        batch_size = self.configs['infer_batch_size']
```
Comment (hunterhector) [inline forte_wrapper/allennlp/allennlp_processors.py:141]: I think we need a bit type annotation here. We haven't set up the mypy pipeline, but we should be using them later. And I think mypy might complaint about some missing typing here, such as `batches`, `inputs`, `results`

**Review by hunterhector — COMMENTED** [review]
I think it'd be nice to setup some type annotations.

File: forte_wrapper/allennlp/allennlp_processors.py (hunk 1 above)
Comment (DianLiI) [reply forte_wrapper/allennlp/allennlp_processors.py]: Checked with mypy, the only thing it complains is `batches` so I've fixed it. All other variables seems to be inferred correctly but added hints for them anyways.

**Review by hunterhector — APPROVED** [review]

File: forte_wrapper/allennlp/allennlp_processors.py [hunk 2]
```diff
@@ -116,6 +117,8 @@ def default_configs(cls):
                 multiple models are loaded, cuda devices are assigned in a
                 round robin fashion. E.g. [0, -1] -> first model uses gpu 0
                 but second model uses cpu.
+            - infer_batch_size: batch size used for model inference.
+                A value <= 0 means no limit.
```
Comment (hunterhector) [inline forte_wrapper/allennlp/allennlp_processors.py:121]: I guess we need to say the size is corresponding to `Sentence` in this case?

Comment (DianLiI) [reply forte_wrapper/allennlp/allennlp_processors.py]: Re-phrased it to: maximum number of sentences passed in as a batch to model's predict function.

**Review by hunterhector — APPROVED** [review]

**Pull request merged**